knowledge_base/Web-Wallets.md

```diff
@@ -0,0 +1,123 @@
+
+    + Within the 'terra' community, only the 'terrastation' and 'terra-walletconnect' wallets are rendered.
+
+    + Within the 'evmos' and 'injective' communities, only 'cosm-metamask' and 'keplr-ethereum' wallets are rendered.
+
+    + If the user doesn't have any of the wallets which are required for the context of the page, then the wallet list is rendered empty, with a `No wallets found` message. The SSO list is still displayed and user can use that to register an account and join a community.
+
+## Sign-in & Onboarding Flow
+
+As of 240601, a new User Onboarding flow is being rolled out; it is toggled by the `userOnboardingEnabled` [feature flag](./Feature-Flags.md).
+
+1. The user begins the authentication flow by clicking  the `Sign Up` or `Create Account` buttons in the header, or else by attempting to perform an auth-gated action.
+
+2. Assuming a user selects the `Create Account` button, the [CreateAccountModal](../packages/commonwealth/client/scripts/views/modals/AuthModal/CreateAccountModal/CreateAccountModal.tsx) is opened.
+
+    1. The user is shown 2 options, `Create a wallet` and `I have a wallet`.
+
+        + If `Create a wallet` is selected, only SSO options are rendered.
+
+        + If `I have a wallet` is selected, only Web3 wallets are rendered.
+
+    2. The user selects from the displayed wallet or SSO options.
+
+        + If the user already has an account linked to the selected wallet or SSO option, then they will be signed into the app once the auth process is complete.
+
+        + If the user does not have an account linked to the selected wallet or SSO option, the [WelcomeOnboardModal](../packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx) is opened once the authentication process completes.
+
+3. Assuming a user clicked on the `Signup` button, the [SignInModal](../packages/commonwealth/client/scripts/views/modals/AuthModal/SignInModal/SignInModal.tsx) is opened.
+
+    + The user can select from any available wallet or SSO option.
+
+    + If the user already has an account linked to the selected wallet or SSO option, they are signed into the app once the authentication process completes.
+
+4. If the user does not have an account linked to the selected wallet or SSO option, the [`AuthTypeGuidanceModal`](../packages/commonwealth/client/scripts/views/modals/AuthModal/AuthTypeGuidanceModal/AuthTypeGuidanceModal.tsx) is opened.
+
+    + If `Create an account` is selected, the `CreateAccountModal` is displayed.
+
+    + If `Sign in another way` is selected, the `SignInModal` is displayed.
+
+```mermaid
+flowchart LR
+    AuthTypeGuidanceModal --> CreateAccountModal
+    AuthTypeGuidanceModal --> SignInModal
+
+    CreateAccountModal --> WelcomeOnboardModal
+```
+
+### Sign-in state
+
+A user's active account is stored in `app.user.activeAccount`, and updated in the [login controller](../packages/commonwealth/client/scripts/controllers/app/login.ts).
```

Comment:
this might be a bit misleading. `app.user.activeAccount` only stores the active "address" in a community, and its not present in non-community pages.


Further context:

- For Communities - If a user has x number of accounts (linked addresses), only 1 of them is `activeAccount`.
- For non-scoped/community pages - we mostly rely on login state ex: `app.isLoggedIn()`. The only case I recall where we have to select a certain address in non-scoped page is when buying community stake from the `/communities` page